src/ResultWriter/ini_faultoutput.f90

```diff
@@ -104,6 +105,19 @@ SUBROUTINE fill_variables_list(DISC, VariableList)
        IF (j.EQ.6.AND.DISC%DynRup%DynRup_out_atPickpoint%OutputMask(3).NE.1) CYCLE
        IF (j.EQ.7.AND.DISC%DynRup%DynRup_out_atPickpoint%OutputMask(4).NE.1) CYCLE
        IF (j.EQ.8.AND.DISC%DynRup%DynRup_out_atPickpoint%OutputMask(4).NE.1) CYCLE
+       IF (j.EQ.9.AND.DISC%DynRup%DynRup_out_atPickpoint%OutputMask(5).NE.1) CYCLE
+       IF (j.EQ.10.AND.DISC%DynRup%DynRup_out_atPickpoint%OutputMask(5).NE.1) CYCLE
+       IF (j.EQ.11.AND.DISC%DynRup%DynRup_out_atPickpoint%OutputMask(5).NE.1) CYCLE
+       IF (j.EQ.12.AND.DISC%DynRup%DynRup_out_atPickpoint%OutputMask(6).NE.1) CYCLE
+       IF (j.EQ.13.AND.DISC%DynRup%DynRup_out_atPickpoint%OutputMask(6).NE.1) CYCLE
+       IF (j.EQ.14.AND.DISC%DynRup%DynRup_out_atPickpoint%OutputMask(7).NE.1) CYCLE
+       IF (j.EQ.15.AND.DISC%DynRup%DynRup_out_atPickpoint%OutputMask(8).NE.1) CYCLE
+       IF (j.EQ.16.AND.DISC%DynRup%DynRup_out_atPickpoint%OutputMask(9).NE.1) CYCLE
+       IF (j.EQ.17.AND.DISC%DynRup%DynRup_out_atPickpoint%OutputMask(10).NE.1) CYCLE
+       IF (j.EQ.18.AND.DISC%DynRup%DynRup_out_atPickpoint%OutputMask(11).NE.1) CYCLE
+       IF (j.EQ.19.AND.DISC%DynRup%DynRup_out_atPickpoint%OutputMask(12).NE.1) CYCLE
+       IF (j.EQ.20.AND.DISC%DynRup%DynRup_out_atPickpoint%OutputMask(12).NE.1) CYCLE
+
```

Comment:
Please rewrite this code in a more sensible way.

Reply:
to be honest I do not know how to do that. What does sensible way mean?

Reply:
More sensible means:
- Less code repetition
- Not checking 20 if statements in each iteration

E.g.

maskCount = (/ 2, 3, 1, 2, 3, 2, 1, 1, 1, 1, 1, 2 /)

k = 1
do i=lbound(maskCount),ubound(maskCount)
  if (DISC%...%OutputMask(i) .EQ. 1)
    do j=0,maskCount(i)-1
       WRITE(VariableList,'(a,a3,a,a1)') TRIM(VariableList_temp),',"',TRIM(VName(k+j)),'"'
    end do
  end if
  k = k + maskCount(i)
end do